server/datastore/mysql/teams.go

```diff
@@ -15,7 +16,11 @@ import (
 
 var teamSearchColumns = []string{"name"}
 
+const teamColumns = `id, created_at, name, description, config`
+
 func (ds *Datastore) NewTeam(ctx context.Context, team *fleet.Team) (*fleet.Team, error) {
+	// We must normalize the name for full Unicode support (Unicode equivalence).
+	team.Name = norm.NFC.String(team.Name)
```

Comment:
You are changing the input params here.
Anything that could go wrong for what called this function?

Reply:
Changing the `team` struct is expected here. We already change it by populating the ID. So, updating the name to match what is in the DB seems correct.